### DISCUSSION TITLE
ScriptReference/RuntimeInitializeOnLoadMethodAttribute

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#209] [Notes]
# ScriptReference/RuntimeInitializeOnLoadMethodAttribute

https://docs.unity3d.com/ScriptReference/RuntimeInitializeOnLoadMethodAttribute.html

**Comment by mvi** [comment]
Note that you should never wrap a `[RuntimeInitializeOnLoadMethod]` in an if def as it will not behave as expected. Instead you can put your if def inside the method body.  
For full details on how this attribute behaves strangely with if defs [see here](https://stackoverflow.com/questions/44655667/runtimeinitializeonload-not-running-with-conditional-compilation).
